script.twittertweets/service.py

```diff
@@ -0,0 +1,141 @@
+SETTINGS_LOC  = REAL_SETTINGS.getAddonInfo('profile')
+ADDON_PATH    = REAL_SETTINGS.getAddonInfo('path').decode('utf-8')
+ADDON_VERSION = REAL_SETTINGS.getAddonInfo('version')
+ICON          = REAL_SETTINGS.getAddonInfo('icon')
+FANART        = REAL_SETTINGS.getAddonInfo('fanart')
+LANGUAGE      = REAL_SETTINGS.getLocalizedString
+
+## GLOBALS ##
+DEBUG       = REAL_SETTINGS.getSetting('Enable_Debugging') == 'true'
+BASE_URL    = 'http://twitter.com/'
+
+def log(msg, level=xbmc.LOGDEBUG):
+    if DEBUG == True:
+        if level == xbmc.LOGERROR:
+            msg += ' ,' + traceback.format_exc()
+        xbmc.log(ADDON_ID + '-' + ADDON_VERSION + '-' + stringify(msg), level)
+        
+def stringify(string):
+    if isinstance(string, list):
+        string = (string[0])
+    elif isinstance(string, (int, float, long, complex, bool)):
+        string = str(string) 
+    
+    if isinstance(string, basestring):
+        if not isinstance(string, unicode):
+            string = unicode(string, 'utf-8')
+        elif isinstance(string, unicode):
+            string = string.encode('ascii', 'ignore')
+        else:
+            string = string.encode('utf-8', 'ignore')
+    return string
+
+def getProperty(str):
+    try:
+        return xbmcgui.Window(10000).getProperty(stringify(str))
+    except Exception,e:
+        log("getProperty, Failed! " + str(e), xbmc.LOGERROR)
+        return ''
+          
+def setProperty(str1, str2):
+    try:
+        xbmcgui.Window(10000).setProperty(stringify(str1), stringify(str2))
+    except Exception,e:
+        log("setProperty, Failed! " + str(e), xbmc.LOGERROR)
+
+def clearProperty(str):
+    xbmcgui.Window(10000).clearProperty(stringify(str))
+   
+class Service():
+    def __init__(self):
+        log('__init__')
+        random.seed()
+        self.userList  = []
+        self.myService = xbmc.Monitor()
+                             
+        for i in range(1,51):
+            self.userList.append((REAL_SETTINGS.getSetting('FEED%d'%i)).replace('@',''))
+
+        while not self.myService.abortRequested():
+            WAIT_TIME = [300,600,900,1800][int(REAL_SETTINGS.getSetting('Wait_Time'))]
+            IGNORE    = REAL_SETTINGS.getSetting('Not_While_Playing') == 'true'
+            if xbmc.Player().isPlayingVideo() == True and IGNORE == True:
+                self.myService.waitForAbort(WAIT_TIME)
+                continue
+            self.userList = filter(None, self.userList)
+            log('userList = ' + str(self.userList))
+            for user in self.userList:
+                self.chkFEED(user)
+            if self.myService.waitForAbort(WAIT_TIME) == True:
+                break
+
+                
+    def testString(self):
+        ''' 
+        gen. 140char mock sentence for skin test
+        '''
+        a = ''
+        for i in range(1,141):
+            a += 'W%s'%random.choice(['',' '])
+        return a[:140]
+        
+
+    def chkFEED(self, user):
```

Comment:
I didn't notice that module... THX I'll take a look for a future update. As it is I'd like to keep beautifulsoup since the script is simple and I hope requires little maintenance. I will definitely consider moving to the API for long term stability if twitter changes their site around.